### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by maov19**
### Implementations
**1. Removed current_user method in ApplicationController.**

**2. Installed and setup devise, adding the following functions:**
       - Can register a new user.
       - Ask for confirmation of email.
       - User logs in with a combination of email and password
       - Can reset password.
       - Hashed password is stored in the database.

**3. Modified the views in devise for registration and login to match the model wireframes.**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,9 +1,14 @@
 class User < ApplicationRecord
+  # has_secure_password
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (bienvenuushindi) [inline app/models/user.rb]: :red_circle: As [per this requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md) Your blog-app should implement the `:confirmable` module. Please add and implement this module as it is one of  the project requirements

requi

File: db/migrate/20230621072445_add_devise_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      # t.datetime :current_sign_in_at
+      # t.datetime :last_sign_in_at
+      # t.string   :current_sign_in_ip
+      # t.string   :last_sign_in_ip
+
+      ## Confirmable
+      # t.string   :confirmation_token
+      # t.datetime :confirmed_at
+      # t.datetime :confirmation_sent_at
+      # t.string   :unconfirmed_email # Only if using reconfirmable
```
Comment (bienvenuushindi) [inline db/migrate/20230621072445_add_devise_to_users.rb]: :red_circle: As [per this requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md) Your blog-app should implement the `:confirmable` module. Please enable and implement this module as it is one of  the project requirements

requi

**Review by bienvenuushindi — CHANGES_REQUESTED** [review]
### Hi Team @maov19  and @edubew ,

Good job implementing most of the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- Github flow is used correctly. ✔
- All linters checks are ok. ✔


## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:


_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: db/migrate/20230621072445_add_devise_to_users.rb (hunk 2 above)
Comment (maov19) [reply db/migrate/20230621072445_add_devise_to_users.rb]: Fixed! Added the confirmable module, as required, both in migration file and models.

File: app/models/user.rb (hunk 1 above)
Comment (maov19) [reply app/models/user.rb]: Fixed! Added the confirmable module, as required, both in migration file and models.

Comment (bienvenuushindi) [reply app/models/user.rb]: Hi @maov19,

Great work! :+1:

Please push your changes to GitHub and request for a re-review from your dashboard.

Thanks,

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Proper implementation of reviewer suggestions

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**